### DISCUSSION TITLE
React 實戰篇 - 留言板

### DISCUSSION THREAD

**Discussion by chishin-lin** [Lidemy/forum#219] [前端相關問題]
老師請教一下：
**React 實戰篇 - 留言板**
下面這個state不太理解他的作用
const [ isLoadingPostMessage , setIsLoadingPostMessage ] = useState(false);

特別是在：
handleFormSubmit這個event function使用(粗體部分)：
 e.preventDefault();
    
**if (isLoadingPostMessage) {
      return;
    }**
    
**setIsLoadingPostMessage(true);**
    fetch.....這裡可以....
     .catch((err) => {  **setIsLoadingPostMessage(false);** }

**Comment by aszx87410** [comment]
第一個作用是這個狀態可以用來決定要不要顯示 loading 的畫面
第二個作用是避免打兩次 api
